pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -117,13 +124,18 @@ public HelixBrokerStarter(PinotConfiguration brokerConf, String clusterName, Str
       brokerHost = _brokerConf.getProperty(CommonConstants.Helix.SET_INSTANCE_ID_TO_HOSTNAME_KEY, false) ? NetUtil
           .getHostnameOrAddress() : NetUtil.getHostAddress();
     }
+
     _brokerId = _brokerConf.getProperty(Helix.Instance.INSTANCE_ID_KEY,
-        Helix.PREFIX_OF_BROKER_INSTANCE + brokerHost + "_" + _brokerConf
-            .getProperty(Helix.KEY_OF_BROKER_QUERY_PORT, Helix.DEFAULT_BROKER_QUERY_PORT));
+        Helix.PREFIX_OF_BROKER_INSTANCE + brokerHost + "_" + inferPort());
 
     _brokerConf.addProperty(Broker.CONFIG_OF_BROKER_ID, _brokerId);
   }
 
+  private int inferPort() {
+    return Optional.ofNullable(_brokerConf.getProperty(Helix.KEY_OF_BROKER_QUERY_PORT)).map(Integer::parseInt)
```

Comment:
Not sure how this ensures that if nothing is configured, we still start without tls on DEFAULT_BROKER_QUERY_PORT (previous behavior).

Reply:
starting on DEFAULT_BROKER_QUERY_PORT is ensured by ListenerConfigUtil. Here, we merely need to figure out a port number to generate an instance id for Helix

Reply:
Do we do that correctly if nothing is configured?

Reply:
This is the same code as on the multi-ingress controller. For completeness sake, I verified the behavior for different scenarios via quickstart:

pinot.broker.client.queryPort only: `Broker_192.168.0.126_8000`
pinot.broker.client.queryPort and https ingress: `Broker_192.168.0.126_8000`
http and https ingress only: `Broker_192.168.0.126_8000`
https ingress only: `Broker_192.168.0.126_8443`
no config: `exception "No value present”`

Reply:
Thanks, I think this exception is ok, since we are the same behavior as the controller. But it is useful to throw the right error message in the exception mentioning which configuration the users should provide.